### DISCUSSION TITLE
续期新业务后打开“我的企业”菜单变成新业务的菜单了

### DISCUSSION THREAD

**Discussion by reamZMX** [zuihou/zuihou-admin-cloud#356] [02.提问] [answered]
# 版本信息 （必填）：
后端使用的是lamp-cloud-datasource-column
前端使用的是lamp-web-pro
JDK 版本(必填) : 8
源码版本(必填)：4.18

# 问题描述：
新建服务成功后，进行了单表的代码生成，当我给用户授权时，打开“我的企业”变成了我生成后的菜单，请问是有什么配置没做吗？
# 报错截图
![image](https://github.com/user-attachments/assets/404f9a70-1ae6-40c8-b2fb-63d90851148f)

# 重现步骤
1. 不给用户续期
![image](https://github.com/user-attachments/assets/fb831b73-0aed-4c5d-b8e7-45309ff3c201)
显示是正常的
![image](https://github.com/user-attachments/assets/b5c69ce8-f74a-46fa-8389-62b9d13afe64)
2.给用户续期后
![image](https://github.com/user-attachments/assets/64a6a0dd-bcca-4fdd-a975-6545d97533fe)
再刷新页面就有问题了
![image](https://github.com/user-attachments/assets/07e65b73-749a-4187-9241-1a919930740a)
菜单数据等都是直接生成的

**Comment by zuihou** [comment, marked as answer]
这是前端菜单显示的问题，2种方案：

1. 导航栏模式中 选用 其他模式
<img width="462" alt="image" src="https://github.com/user-attachments/assets/565a3293-5408-4d43-a295-a67c10188421">
2. 更新前端的代码，将“我的企业”隐藏。
<img width="1542" alt="image" src="https://github.com/user-attachments/assets/3dcdd0b6-6eed-4ef1-b17c-36b28b1cd970">

**Reply by reamZMX** [reply]
当我升级版本后，出现了这样一个问题：
target中的yml已经被成功替换为dev环境 但是当我运行application时却获取不到参数
![image](https://github.com/user-attachments/assets/5d0fa167-59ba-40a2-8fff-78922d6812e3)
nacos是正常启动的
顺便问一下 4.21.2算最新版吗？会有这个问题吗？
感谢

**Reply by zuihou** [reply]
<img width="1176" alt="image" src="https://github.com/user-attachments/assets/a3c8e911-60d4-4e45-bba7-446a52aa7823">


看下这里有没有替换成功， 没成功就先 mvn clean， 然后重新 mvn install

**Reply by reamZMX** [reply]
![image](https://github.com/user-attachments/assets/757b4ee6-959e-406c-8418-3218cd1deb3c)
替换成功了的

**Comment by zuihou** [comment]
不能加引号！

<img width="1070" alt="image" src="https://github.com/user-attachments/assets/11ed6b7b-2709-4110-b61c-2aecb9eb0d7d">

<img width="1311" alt="image" src="https://github.com/user-attachments/assets/a33471df-53e9-4b6f-89b7-d7a3be384d9a">

**Reply by reamZMX** [reply]
当我不加引号时会报错：
![image](https://github.com/user-attachments/assets/f7bc3b02-a7cc-4561-81d8-3015acecefe0)
加引号是可以被读取的  当我没有升级版本的时候

**Reply by zuihou** [reply]
不加引号，编译项目(关键是编译！  mvn install)，让 target 下面的 bootstrap.yml 替换成功。


`@xxx@` 占位符的参数，都是在项目编译期间会替换的，没有成功替换，说明你的项目还存在问题。

**Reply by zuihou** [reply]
<img width="1128" alt="image" src="https://github.com/user-attachments/assets/7962efb9-f66b-477e-a8f7-db6915c069e0">

**Comment by zuihou** [comment]
关于 `@xxx@` 参数， 可以看： https://github.com/dromara/lamp-cloud/discussions/49

**Comment by reamZMX** [comment]
执行版本升级的时候请关闭IDEA，我升级版本没关闭IDEA，导致我IDEA项目配置出现了问题，运行application时没读取到配置，当我在运行配置选择工作目录时：
![image](https://github.com/user-attachments/assets/a93d6546-288d-445f-a5cc-55f10c84228c)
当我指定到对应模块的target的classes目录即可运行成功
但是我之前是不需要配置的，然后经过一番摸索，发现在项目结构的模块中，所有的输出路径变成了相同的：
![image](https://github.com/user-attachments/assets/7ab901ec-3125-472a-a389-2a10ffa0df71)
所以读取不到配置
关于单引号，我电脑环境确实是需要单引号的：
![image](https://github.com/user-attachments/assets/022d9bcb-2ada-4fc5-a565-d33c5dd1c732)
只有当我全文替换带有单引号的字符串yml文件才会被正确解析

**Reply by zuihou** [reply]
👍🏻
